### PR TITLE
feat(medusa, stock-location-next): add list-stock-locations endpoint to api-v2

### PR DESCRIPTION
**Pull request by pKorsholm**

### CODE REVIEW

**Review by carlos-r-l-rodrigues — APPROVED** [review]

**Review by sradevski — APPROVED** [review]

File: .changeset/six-flies-dress.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@medusajs/core-flows": patch
+"@medusajs/medusa": patch
```
Comment (adrien2p) [inline .changeset/six-flies-dress.md]: q: why do we have 2 changeset and the core flows in it?

File: packages/medusa/src/api-v2/admin/stock-locations/route.ts [hunk 2]
```diff
@@ -30,3 +30,29 @@ export const POST = async (
...
+export const GET = async (req: MedusaRequest, res: MedusaResponse) => {
+  const remoteQuery = req.scope.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
+
+  const query = remoteQueryObjectFromString({
+    entryPoint: "stock_locations",
+    variables: {
+      filters: req.filterableFields,
+      order: req.listConfig.order,
+      skip: req.listConfig.skip,
+      take: req.listConfig.take,
```
Comment (adrien2p) [inline packages/medusa/src/api-v2/admin/stock-locations/route.ts]: todo: ...req.remoteQueryConfig.pagination

File: packages/medusa/src/api-v2/admin/stock-locations/route.ts [hunk 3]
```diff
@@ -30,3 +30,29 @@ export const POST = async (
...
+    },
+    fields: req.remoteQueryConfig.fields,
+  })
+
+  const { rows: stock_locations, metadata } = await remoteQuery({
+    ...query,
+  })
```
Comment (adrien2p) [inline packages/medusa/src/api-v2/admin/stock-locations/route.ts]: todo: await remoteQuery(query)

**Review by adrien2p — APPROVED** [review]
🔥 few changes

**Pull request merged**